dags/bqetl_artifact_deployment.py

```diff
@@ -56,11 +58,26 @@
 
 tags = [Tag.ImpactTier.tier_1]
 
+params = {
+    "generate_sql": Param(
+        default=False,
+        type="boolean",
+        description="Run sql generation before each publish task",
+    ),
+}
+
+# renders generate sql command if params.generate_sql is true, else empty string
+generate_sql_cmd_template = (
+    "{{ 'script/bqetl generate all --use-cloud-function=false && ' "
+    "if params.generate_sql else '' }}"
+)
+
 
-def check_for_queued_runs(dag_id: str) -> bool:
+def check_for_queued_runs(dag_id: str, generate_sql: bool) -> bool:
```

Comment:
The param will get passed as a string unless you specify `render_template_as_native_obj`.
```suggestion
def check_for_queued_runs(dag_id: str, generate_sql: str) -> bool:
```